### PR TITLE
Create test-coverage.yml action

### PR DESCRIPTION
**Pull request by fazzatti**

### CODE REVIEW

File: .github/workflows/test-coverage.yml [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+name: Node.js CI
+
+on:
+  workflow_dispatch:
+  workflow_call:
+  push:
+    branches: [ "develop" ]
+  pull_request:
+    branches: [ "develop" ]
```
Comment (monteiro-alan) [inline .github/workflows/test-coverage.yml:12]: What about main?

File: .github/workflows/test-coverage.yml [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-node@v3
+      with:
+        node-version: 16
```
Comment (monteiro-alan) [inline .github/workflows/test-coverage.yml:21]: Is Node 16 the only version that we need to test?

File: .github/workflows/test-coverage.yml [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+  push:
+    branches: [ "develop" ]
+  pull_request:
+    branches: [ "develop" ]
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
```
Comment (monteiro-alan) [inline .github/workflows/test-coverage.yml:18]: I like to always use this structure to keep standards steps. This is not an obligation, just a suggestion!
```
      - name: Checkout
        uses: actions/checkout@v3
```

The ideia is to have a name for each steps, to make it clear what the steps is doing when the job runs.

File: .github/workflows/test-coverage.yml [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    - run: npm ci
+    - run: npm run build --if-present
+    
+    - name: Run tests with coverage
+    - run: npm test
+
+    - name: Code Coverage Report
+      uses: irongut/CodeCoverageSummary@v1.3.0
+      with:
+        filename: ./src/coverage/cobertura-coverage.xml
```
Comment (monteiro-alan) [inline .github/workflows/test-coverage.yml:33]: Just to make sure, did you tested if the report is being generated in this folder?

File: .github/workflows/test-coverage.yml [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    steps:
+    - uses: actions/checkout@v3
+    - uses: actions/setup-node@v3
+      with:
+        node-version: 16
+    
+    - name: Build
```
Comment (monteiro-alan) [inline .github/workflows/test-coverage.yml:23]: Each `-` is a step, so this is structure is incorrect

**Review by monteiro-alan — CHANGES_REQUESTED** [review]

File: .github/workflows/test-coverage.yml (hunk 1 above)
Comment (fazzatti) [reply .github/workflows/test-coverage.yml:12]: Just wanted to give it a try with only dev first as we're only pushing to main near a new version release, the idea is to add main next as well.

File: .github/workflows/test-coverage.yml (hunk 4 above)
Comment (fazzatti) [reply .github/workflows/test-coverage.yml:33]: Yeah, just ran locally and double checked! 👍

**Pull request merged**